### ISSUE TITLE
Windows Sub-System Linus (WSL) Ubuntu Error

### ISSUE THREAD

**Issue by adamparish** [mobilecoinofficial/mobilecoin#83]
**Getting the following socket error when trying to run the Linux testnet client:**

```
asp@Surface-Laptop:~/mobilecoin-testnet-linux$ ./mobilecoin-testnet.sh
Daemon is starting up
E0423 14:31:50.556821900     402 socket_utils_common_posix.cc:313] setsockopt(TCP_USER_TIMEOUT) Protocol not available
E0423 14:31:52.560275900     406 socket_utils_common_posix.cc:313] setsockopt(TCP_USER_TIMEOUT) Protocol not available
Unable to connect to mobilecoind on 127.0.0.1:4444.
Are you sure it is running and accepting connections?

The error was: RpcFailure: 14-UNAVAILABLE failed to connect to all addresses
Terminated
```

**Environment Details:**

```
asp@Surface-Laptop:~/mobilecoin-testnet-linux$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.4 LTS
Release:        18.04
Codename:       bionic
```

**Comment by eranrund** [comment]
Is there a `mobilecoind.log` file in that directory?
If so, could you please share its contents?

Thanks!

**Comment by adamparish** [comment]
Indeed! Thanks for taking a look. I suspect this issue has something to do with the minimal nature of WSL on Windows 10.

[mobilecoind.log](https://github.com/mobilecoinofficial/mobilecoin/files/4524685/mobilecoind.log)

**Comment by eranrund** [comment]
No problem!

So, yeah, the code that produces that is https://github.com/mobilecoinofficial/mobilecoin/blob/49076e3ff46ca6d489fb60eacf55fb36a33a207d/mobilecoind/src/bin/main.rs#L152 which has very little logic in it other than making some calls to `lmdb`. If that fails with this rather weird image thats indeed where I would place my bet on.

I don't have experience with Windows/WSL but I believe your suspicion is correct and that this is some weird WSL issue. https://github.com/microsoft/WSL/issues/3451 might be relevant.

**Comment by adamparish** [comment]
Can we leave this open for a couple of days. I'd like to post the link on the Community Forum to see if anyone else tries to get WSL working.

**Comment by adamparish** [comment]
Created a post on the MobileCoin Community Forum to see if other Windows 10 WSL users might have a suggestion. https://community.mobilecoin.com/t/wsl-ubuntu-linux/149
